lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -657,16 +657,23 @@ class DxilFinalizeModule : public ModulePass {
           // In spite of the <=1 assert above, loop here in case the assumption is wrong
           for (auto II = U->user_begin(), EE = U->user_end(); II != EE;) {
             User *UU = *II++;
-            BranchInst *BI = cast<BranchInst>(UU);
-            Function *F = BI->getParent()->getParent();
+            Instruction *I = dyn_cast<Instruction>(UU);
+            if (!I) continue;
+            Function *F = I->getParent()->getParent();
             ICmpInst *Cmp = DxBreakCmpMap.lookup(F);
             if (!Cmp) {
               BasicBlock &EntryBB = F->getEntryBlock();
               LoadInst *LI = new LoadInst(Gep, nullptr, false, EntryBB.getTerminator());
               Cmp = new ICmpInst(EntryBB.getTerminator(), ICmpInst::ICMP_EQ, LI, llvm::ConstantInt::get(i32Ty,0));
               DxBreakCmpMap.insert(std::make_pair(F, Cmp));
             }
-            BI->setCondition(Cmp);
+            // Determine which operand uses the boolean output and replace it
+            for (unsigned i = 0; i < I->getNumOperands(); i++) {
```

Comment:
How about `UU->replaceUsesOfWith()` here?

Reply:
Ah didn't see this. that's a good suggestion

Reply:
Right, this sounds much better.  It also eliminates the need for `I`.  If `F` is retrieved from `CI` instead, then you never even need to get `I` right?

Reply:
In fact, I don't think we need to iterate users of the CallInst, do we?

Reply:
we discussed this before and I told you I did it out of fear and that I was willing to change it. Shall I?

Reply:
If you use ReplaceUsesWith(), you don't need this.

Reply:
The whole becomes something like:
```
        // Replace all uses of dx.break with references to the constant global
        for (auto I = BreakFunc->user_begin(), E = BreakFunc->user_end(); I != E;) {
          User *U = *I++;
          CallInst *CI = cast<CallInst>(U);
          Function *F = CI->getParent()->getParent();
          ICmpInst *Cmp = DxBreakCmpMap.lookup(F);
          if (!Cmp) {
            BasicBlock &EntryBB = F->getEntryBlock();
            LoadInst *LI = new LoadInst(Gep, nullptr, false, EntryBB.getTerminator());
            Cmp = new ICmpInst(EntryBB.getTerminator(), ICmpInst::ICMP_EQ, LI, llvm::ConstantInt::get(i32Ty,0));
            DxBreakCmpMap.insert(std::make_pair(F, Cmp));
          }
          CI->replaceUsesWith(Cmp);
          CI->eraseFromParent();
        }
```

Reply:
replaceUsesWith doesn't exist. I think you're thinking of replaceAllUsesWith, nevertheless, I think replaceUsesOfWith is more appropriate. That's what I have used.